src/main/java/uk/nhs/hee/tis/trainee/forms/api/util/AuthTokenUtil.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2020 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.forms.api.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+@Slf4j
+public class AuthTokenUtil {
+
+  private static final String TIS_ID_ATTRIBUTE = "custom:tisId";
+
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  private AuthTokenUtil() {
```

Comment:
if we're dealing with tokens, should we not validate them against the secret?

Reply:
Wasn't sure about this when I was doing it to be honest. The expectation is that the service itself will always be private and public access is only via API Gateway. Given the API Gateway will authorize the token I've assumed, perhaps naïvely, that the token can be trusted within this service to keep things fairly simple.

It does rely on the service being kept private and only ever accessible through the API Gateway, I suppose there is room for error there.

Reply:
if the api gateway does the jwt validation then I don't see any issue with this. last thing you want is to duplicate this validation code everywhere in other services - yay for shared modules